annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastPackageResolver.java

```diff
@@ -53,4 +53,66 @@ private static String getParentPackage(String packageString) {
         }
         return packageString.substring(0, lastDotIndex);
     }
+
+    public void addFieldToFastType(Element element) {
+        FastFieldElement fastField = new FastFieldElement(element);
+        FastTypeElement fastType = getFastTypeForField(element);
+        if (fastType != null) {
+            fastType.addFastField(fastField);
+        } // otherwise ignore @FastFields if class is not annotated with @FastType
+    }
+
+    private FastTypeElement getFastTypeForField(Element element) {
+        Element ownerFastType = element.getEnclosingElement();
+        for (FastPackageElement fastPackage : cache.values()) {
+            for (FastTypeElement fastType : fastPackage.getFastTypes()) {
+                if (fastType.getElement().equals(ownerFastType)) {
+                    return fastType;
+                }
+            }
+        }
+        return null;
+    }
+
+    void inheritFields() {
+        HashSet<FastTypeElement> fastTypes = getAllFastTypes();
+        for (FastTypeElement fastType : fastTypes) {
+            Element currentType = fastType.getElement();
+            while (true) {
+                currentType = getSuperClass(currentType);
+                if (currentType == null) {
+                    break;
+                }
+                ArrayList<FastFieldElement> fastFieldElements = new ArrayList<>(getFastFieldsFor(currentType));
+                fastFieldElements.stream()
+                    .filter(fastField -> !isOverrideMethod(fastType.getFastFields(), fastField.getMethodName()))
+                    .forEach(fastType::addFastField);
+            }
+        }
+    }
+
+    private boolean isOverrideMethod(List<FastFieldElement> fastFields, String methodName) {
+        return fastFields.stream().anyMatch(fastField -> fastField.getMethodName().equals(methodName));
+    }
+
+    private HashSet<FastTypeElement> getAllFastTypes() {
+        HashSet<FastTypeElement> fastTypes = new HashSet<>();
+        cache.values().forEach(fastPackage -> fastTypes.addAll(fastPackage.getFastTypes()));
+        return fastTypes;
+    }
+
+    private ArrayList<FastFieldElement> getFastFieldsFor(Element superClass) {
+        ArrayList<FastFieldElement> fastFields = new ArrayList<>();
+        HashSet<FastTypeElement> allFastTypes = getAllFastTypes();
+        allFastTypes.stream().filter(fastType -> fastType.getElement().equals(superClass))
```

Comment:
What if it's super-super class?

Reply:
Class can not have @FastType annotation but still be super class and have @FastField annotations on its members